checker/src/main/java/org/checkerframework/checker/calledmethods/CalledMethodsChecker.java

```diff
@@ -0,0 +1,96 @@
+ * from being instantiated incompletely.
+ */
+@SuppressWarningsPrefix({
+    // Preferred checkername.
+    "calledmethods",
+    "called.methods",
+    // Deprecated checkernames, supported for backward compatibility.
+    "builder",
+    "object.construction",
+    "objectconstruction"
+})
+@SupportedOptions({
+    CalledMethodsChecker.USE_VALUE_CHECKER,
+    CalledMethodsChecker.COUNT_FRAMEWORK_BUILD_CALLS,
+    CalledMethodsChecker.DISABLED_FRAMEWORK_SUPPORTS,
+})
+@StubFiles({"DescribeImages.astub", "GenerateDataKey.astub"})
+public class CalledMethodsChecker extends BaseTypeChecker {
+
+    /**
+     * If this option is supplied, count the number of analyzed calls to build() in supported
+     * frameworks and print it when analysis is complete. Useful for collecting metrics.
+     */
+    public static final String COUNT_FRAMEWORK_BUILD_CALLS = "countFrameworkBuildCalls";
+
+    /**
+     * This option can be used to disable the support (and therefore the automated checking of) code
+     * that uses the given frameworks. Useful when a user **only** wants to enforce specifications
+     * on custom builder objects (such as the AWS SDK examples).
+     */
+    public static final String DISABLED_FRAMEWORK_SUPPORTS = "disableFrameworkSupports";
+
+    /**
+     * The key for the {@link #DISABLED_FRAMEWORK_SUPPORTS} option to disable Lombok support. Use it
+     * via {@code -AdisableFrameworkSupports=LOMBOK}.
+     */
+    public static final String LOMBOK_SUPPORT = "LOMBOK";
+
+    /**
+     * The key for the {@link #DISABLED_FRAMEWORK_SUPPORTS} option to disable AutoValue support. Use
+     * it via {@code -AdisableFrameworkSupports=AUTOVALUE}.
+     */
+    public static final String AUTOVALUE_SUPPORT = "AUTOVALUE";
+
+    /**
+     * If this option is supplied, use the Value Checker to reduce false positives when analyzing
+     * calls to the AWS SDK.
+     */
+    public static final String USE_VALUE_CHECKER = "useValueChecker";
+
+    /**
+     * The number of calls to build frameworks supported by this invocation, if the {@link
+     * #COUNT_FRAMEWORK_BUILD_CALLS} option was supplied.
+     */
+    int numBuildCalls = 0;
+
+    @Override
+    protected LinkedHashSet<Class<? extends BaseTypeChecker>> getImmediateSubcheckerClasses() {
+        LinkedHashSet<Class<? extends BaseTypeChecker>> checkers =
+                super.getImmediateSubcheckerClasses();
+        checkers.add(ReturnsReceiverChecker.class);
+
+        // BaseTypeChecker#hasOption calls this method (so that all subcheckers' options are
+        // considered),
+        // so the processingEnvironment must be checked for the option directly.
+        if (this.processingEnv.getOptions().containsKey(USE_VALUE_CHECKER)
+                || this.processingEnv
+                        .getOptions()
+                        .containsKey(this.getClass().getSimpleName() + "_" + USE_VALUE_CHECKER)) {
```

Comment:
There is already a comment here explaining why: `BaseTypeChecker#hasOption` calls `getImmediateSubcheckers`, so using the CF's standard option code doesn't work. We have to implement the logic ourselves (for just this option) here, because the effect of the option is to enable a new subchecker.

It might be possible to refactor the logic for options so that this hack can be avoided, but I'd prefer to do that separately (if we do it at all, which I don't see much reason to - it is quite rare for a subchecker to be guarded by an option flag).